src/Dapr.AspNetCore/CloudEventsMiddleware.cs

```diff
@@ -83,17 +95,29 @@ private async Task ProcessBodyAsync(HttpContext httpContext, string charSet)
             string contentType;
 
             // Check whether to use data or data_base64 as per https://github.com/cloudevents/spec/blob/v1.0.1/json-format.md#31-handling-of-data
-            var isDataSet = json.TryGetProperty("data", out var data);
-            var isBinaryDataSet = json.TryGetProperty("data_base64", out var binaryData);
+            // Get the property names by OrdinalIgnoreCase comparison to support case insensitive JSON as the Json Serializer for AspCore already supports it by default.
+            var jsonPropNames = json.EnumerateObject().ToArray();
+
+            var dataPropName = jsonPropNames
+                .Select(d => d.Name)
+                .FirstOrDefault(d => d.Equals("data", StringComparison.OrdinalIgnoreCase)) ?? "";
```

Comment:
Would it make sense to make `"data"` and `"data_base64"` constants as they're used in multiple places?

Reply:
yeah definitely, ill create a class with the constants or something like that